### PR TITLE
Adds migration guide for material localized Strings

### PR DESCRIPTION
**Pull request by chunhtai**
https://github.com/flutter/flutter/pull/124711

## Presubmit checklist
- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by khanhnwin** [comment]
Thanks @chunhtai! Would you mind if we moved those dart code samples to a project in the examples/ directory so that we can import them as snippets?

**Comment by chunhtai** [comment]
I am not sure how this would work, some of the code in the snippet may be broken once the change is landed

**Comment by chunhtai** [comment]
a friendly bump.

File: src/release/breaking-changes/material-localized-strings.md
Comment (sfshaza2) [inline src/release/breaking-changes/material-localized-strings.md:26]:
```suggestion
If you use these strings in your code,
you can access them from `WidgetsLocalizations`instead.
```

File: src/release/breaking-changes/material-localized-strings.md
Comment (sfshaza2) [inline src/release/breaking-changes/material-localized-strings.md:42]:
```suggestion
If you override `MaterialLocalizations` or `WidgetsLocalizations`,
make sure to remove the translations from the `MaterialLocalizations`
subclass and move them to the `WidgetsLocalizations` subclass.
```

File: src/release/breaking-changes/material-localized-strings.md
Comment (sfshaza2) [inline src/release/breaking-changes/material-localized-strings.md:69]: @chunhtai, can you fill this TBD out?

File: src/release/breaking-changes/material-localized-strings.md
Comment (sfshaza2) [inline src/release/breaking-changes/material-localized-strings.md:57]: Add "Code after migration" text here, and complete the two resulting Dart code blocks.

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
@chunhtai, a few changes. Also, @khanhnwin, can you help @chunhtai with the code snippets, or maybe we don't need it? I defer to you.

**Comment by chunhtai** [comment]
a friendly bump

File: src/release/breaking-changes/material-localized-strings.md
Comment (khanhnwin) [inline src/release/breaking-changes/material-localized-strings.md:57]: Does this class need to remain here after we've moved the reorderItemRight getter?

**Review by khanhnwin — COMMENTED** [review]
Let's land this and then I can move the code out to an examples project!

**Review by khanhnwin — APPROVED** [review]

**Comment by khanhnwin** [comment]
Good to land if it looks good to you @sfshaza2

**Review by sfshaza2 — APPROVED** [review]
lgtm!

**Pull request merged**
